web/src/upload_widget.ts

```diff
@@ -1,19 +1,64 @@
-import {$t} from "./i18n";
+import {Uppy} from "@uppy/core";
+import Dashboard from "@uppy/dashboard";
+import type {DashboardOptions} from "@uppy/dashboard";
+import type {ImageEditorOptions} from "@uppy/image-editor";
+import ImageEditor from "@uppy/image-editor";
+import $ from "jquery";
+
+import * as dialog_widget from "./dialog_widget";
+import {$t, $t_html} from "./i18n";
+import * as loading from "./loading";
+import {any_active} from "./modals";
+import * as settings_data from "./settings_data";
+
+const supported_types = ["image/jpeg", "image/png", "image/gif", "image/tiff"];
+
+const default_max_file_size = 5;
+
+let uppy: Uppy;
+let edited_file: File;
+let original_file: File;
+// This ID is used to remove file from Uppy dashboard.
+// In order to removed the form that is created each time an image is uploaded.
+let fileID: string;
+
+export function get_edited_file(): File {
+    return edited_file;
+}
 
 export type UploadWidget = {
     clear: () => void;
     close: () => void;
 };
 
 export type UploadFunction = (
-    $file_input: JQuery<HTMLInputElement>,
+    $file_input: {files: File[]},
```

Comment:
I see that you have used `$file_input` as an object with `files` being an array of `File`. But in your code, this array contains only a single File. So, any reason why this is updated from `JQuery<HTMLInputElement>`?